tests/Unit/AMDGPU/register-control.cpp

```diff
@@ -1,6 +1,5 @@
 // RUN: %hc %s -o %t.out -Xlinker -dump-llvm -Xlinker -dump-dir=%T --amdgpu-target=gfx803
 // RUN: %llvm-dis %T/dump-gfx803.opt.bc -f -o - | %FileCheck %s
-// RUN: %t.out
```

Comment:
should we remove `--amdgpu-target=gfx803` and rely on the enumerator to decide on the arch?

Reply:
@scchan I don't like to make `hcc` tests depend on `rocm_agent_enumerator`, as it's another tool not owned/built alongside with `hcc`.  Plus it appears the source code of `rocm_agent_enumerator` isn't posted online so it makes porting it to another architecture hard.

Reply:
maybe we should define a new substitution pattern in lit.cfg such that it would replace it with a string including all the GPU architectures?  When we have to add a new architecture in the future, we'll just have to put that in one file.

https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/tests/lit.cfg#L142

Reply:
@scchan I've thought that, but problem with `rocm_agent_enumerator` is very peculiar:

- unlike most other tools specified in `lit.cfg`, it is NOT owned by `hcc`, so we don't usually know its existence or whereabout.
- unlike `c++filt` which is normally placed under `$PATH`, `rocm_agent_enumerator` is usually installed at `/opt/rocm/bin`, which may usually not be in `$PATH`
- and on a new architecture, `rocm_agent_enumerator` may not even install on `/opt/rocm/bin` and could be altered by `ROCM_ROOT` env var

due to these peculiarities of `rocm_agent_enumerator`, I don't feel like adding an extra dependency to it in `hcc` lit tests.

Reply:
my previous comment was about creating something like:

`%target_all_gpus == "--amdgpu-target=gfx701 --amdgpu-target=gfx801 --amdgpu-target=gfx802   --amdgpu-target=gfx900" `

in lit.cfg so that someone could use %target_all_gpus in the unit tests.
It's not about using the enumerator.

Reply:
makes sense. let me revise this PR.